tools/isobuild/compiler-plugin.js

```diff
@@ -1394,8 +1397,19 @@ export class PackageSourceBatch {
     if (canCache) {
       LINKER_CACHE.set(cacheKey, ret);
       if (cacheFilename) {
+        const cacheKeySuffixRegex = new RegExp('_' + cacheKeySuffix, 'g');
+        const oldCacheFilename =
+          cacheFilename.replace(cacheKeySuffixRegex, '_*');
         // Write asynchronously.
-        Fiber(() => files.writeFileAtomically(cacheFilename, retAsJSON)).run();
+        Fiber(() => {
+          try {
+            files.rm_recursive(oldCacheFilename);
```

Comment:
Sorry, I spoke too soon @benjamn - as currently implemented, we need the remove to happen before writing the new cache file; otherwise if the remove fires after writing the new file, it  would effectively wipe the new file as well (since both the old and new files are prefixed with the same hash, and we're removing by that hash). That being said, maybe we're okay with the remove being synchronous in this case, since both the remove and write are happening in a fiber anyways? The remove will hold the write up but the write should still be done well before the created cache file is needed by a rebuild.